modules/gateway/peers.go

```diff
@@ -198,11 +204,14 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 	var remoteVersion string
 	if err := encoding.ReadObject(conn, &remoteVersion, maxAddrLength); err != nil {
 		return err
-	} else if remoteVersion == "reject" {
-		return errors.New("peer rejected connection")
 	}
 	// decide whether to accept this version
-	if build.VersionCmp(remoteVersion, "0.3.3") < 0 {
+	if remoteVersion == "reject" {
+		return errors.New("peer rejected connection")
+	}
+	// Reject peers < v0.4.0 as the previous version is v0.3.3 which is
+	// pre-hardfork.
+	if build.VersionCmp(remoteVersion, "0.4.0") < 0 {
 		conn.Close()
 		return errors.New("unacceptable version: " + remoteVersion)
```

Comment:
though it's not in the diff, I feel like it's close enough to code being changed that tests should be written to cover this.